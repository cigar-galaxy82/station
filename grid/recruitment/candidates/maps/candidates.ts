/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';
import { readFileSync } from 'fs';

import { buildSuperfaceTest } from '../../../test-config';

const sampleCandidate = {
  name: 'John Doe',
  firstName: 'John',
  lastName: 'Doe',

  email: 'john.doe@fakemail.com',
  phone: '1-859-557-6573',
  address: '25772 Gustave Shore, Iowa, USA',

  education: [
    {
      degree: 'MBA',
      school: 'University of Pennsylvania',
      fieldOfStudy: null,
      startedAt: '2008-03-01',
      endedAt: '2011-03-30',
    },
    {
      degree: 'B.S.',
      school: 'University of Chicago',
      fieldOfStudy: 'Marketing Communication & Economics',
      startedAt: '2004-09-01',
      endedAt: '2007-03-30',
    },
  ],

  workExperience: [
    {
      position: 'Sales Director',
      summary: 'Summary of a work experience at Test Company',
      company: 'Test Company',
      industry: 'Telecommunications',
      current: false,
      startedAt: '2011-03-01',
      endedAt: '2014-03-30',
    },
  ],

  cv: {
    name: 'cv-sample.pdf',
    data: readFileSync('./grid/recruitment/candidates/cv-sample.pdf', {
      encoding: 'base64',
    }),
  },

  links: [
    {
      name: 'twitter',
      url: 'https://twitter.com/candidate.username',
    },
    {
      name: 'Portfolio',
      url: 'https://url.to.portfolio',
    },
  ],
};

export const candidatesTest = (
  provider: string,
  jobIds: { valid: string; invalid: string },
  options?: RecordingProcessOptions
): void => {
  describe(`recruitment/candidates/${provider}`, () => {
    describe('CreateCandidate', () => {
      let superface: SuperfaceTest;

      beforeAll(() => {
        jest.setTimeout(10000);
        superface = buildSuperfaceTest({
          profile: 'recruitment/candidates',
          provider,
          useCase: 'CreateCandidate',
        });
      });

      describe('when specified job does exist', () => {
        it('performs correctly', async () => {
          const result = await superface.run(
            {
              input: {
                jobId: jobIds.valid,
                ...sampleCandidate,
              },
            },
            options
          );

          expect(result.isOk()).toBeTruthy();
          expect(result).toMatchSnapshot();
        });
      });

      describe('when specified job does not exist', () => {
        it('returns error', async () => {
          await expect(
            superface.run(
              {
                input: {
                  jobId: jobIds.invalid,
                  firstName: 'Demo',
                  lastName: 'Testing',
                  email: 'demo_testing@fakemail.com',
                },
              },
              options
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
