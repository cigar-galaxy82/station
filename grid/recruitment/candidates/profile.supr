"""
Manage Candidates 
Fetch, list and create candidates in ATS (Applicant Tracking System).

@deprecated 
Use the recruitment/create-candidate or other profiles instead.
"""
name = "recruitment/candidates"
version = "1.0.0"

