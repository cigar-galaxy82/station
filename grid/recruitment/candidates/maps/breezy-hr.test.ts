import { candidatesTest } from './candidates';

candidatesTest(
  'breezy-hr',
  {
    valid: 'f1cfc5572a5701',
    invalid: '12345ab789d1dd',
  },
  {
    valid: '1e5d0dfd2b6501',
    invalid: '12345ab789d1dd',
  }
);
