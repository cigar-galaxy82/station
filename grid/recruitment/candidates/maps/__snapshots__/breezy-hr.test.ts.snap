// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recruitment/candidates/breezy-hr CreateCandidate when specified job does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "id": "b23ca444e77801",
    "jobId": "f1cfc5572a5701",
    "rateLimit": Object {
      "bucket": "breezy-hr",
      "remainingRequests": 97,
      "remainingRequestsPercentage": 97,
      "resetTimestamp": 1670421590,
      "totalRequests": 100,
    },
  },
}
`;

exports[`recruitment/candidates/breezy-hr CreateCandidate when specified job does not exist returns error 1`] = `
Err {
  "error": "MappedError: Expected error
Properties: {
  \\"title\\": \\"Unknown error\\",
  \\"detail\\": {
    \\"type\\": \\"addCandidateError\\",
    \\"message\\": \\"there was an unknown error creating candidate.\\"
  },
  \\"code\\": \\"UnknownError\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"breezy-hr\\",
    \\"totalRequests\\": 100,
    \\"remainingRequests\\": 95,
    \\"remainingRequestsPercentage\\": 95,
    \\"resetTimestamp\\": 1670421590
  }
}
AST Path: definitions[0].statements[1].statements[0]
Original Map Location: Line 19, column 5",
}
`;

exports[`recruitment/candidates/breezy-hr GetStageChanges when specified candidate does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "stages": Array [
      Object {
        "createdAt": "2022-12-14T11:23:18.812Z",
        "current": true,
        "id": "applied",
        "name": "Applied",
      },
      Object {
        "createdAt": "2022-12-14T11:21:30.696Z",
        "current": false,
        "id": "feedback",
        "name": "Feedback",
      },
      Object {
        "createdAt": "2022-12-14T11:19:32.247Z",
        "current": false,
        "id": "feedback",
        "name": "Feedback",
      },
      Object {
        "createdAt": "2022-12-14T11:15:50.674Z",
        "current": false,
        "id": "feedback",
        "name": "Feedback",
      },
      Object {
        "createdAt": "2022-12-14T11:15:29.811Z",
        "current": false,
        "id": "feedback",
        "name": "Feedback",
      },
    ],
  },
}
`;

exports[`recruitment/candidates/breezy-hr GetStageChanges when specified candidate does not exist returns error 1`] = `
Err {
  "error": "MappedHTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Unknown error\\",
  \\"detail\\": {
    \\"type\\": \\"getStreamError\\",
    \\"message\\": \\"there was an unknown error getting candidate stream.\\"
  },
  \\"code\\": \\"UnknownError\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"breezy-hr\\",
    \\"totalRequests\\": 100,
    \\"remainingRequests\\": 98,
    \\"remainingRequestsPercentage\\": 98,
    \\"resetTimestamp\\": 1671018876
  }
}
AST Path: definitions[1].statements[1].responseHandlers[1].statements[1]
Original Map Location: Line 68, column 7",
}
`;
